mesonbuild/dependencies/misc.py

```diff
@@ -666,3 +667,52 @@ def tool_finish_init(ctdep):
     @staticmethod
     def get_methods():
         return [DependencyMethods.PKGCONFIG, DependencyMethods.CONFIG_TOOL]
+
+
+class ShadercDependency(ExternalDependency):
+
+    def __init__(self, environment, kwargs):
+        super().__init__('shaderc', environment, None, kwargs)
+
+        static_lib = 'shaderc_combined'
+        shared_lib = 'shaderc_shared'
+
+        libs = [shared_lib, static_lib]
+        if self.static:
+            libs.reverse()
+
+        cc = self.get_compiler()
+
+        for lib in libs:
+            self.link_args = cc.find_library(lib, environment, [])
+            if self.link_args is not None:
+                self.is_found = True
+
+                if self.static and lib != static_lib:
+                    mlog.warning('Static library {!r} not found for dependency {!r}, may '
+                                 'not be statically linked'.format(static_lib, self.name))
+
+                break
+
+    def log_tried(self):
+        return 'system'
+
+    @classmethod
+    def _factory(cls, environment, kwargs):
+        methods = cls._process_method_kw(kwargs)
+        candidates = []
+
+        if DependencyMethods.PKGCONFIG in methods:
+            candidates.append(functools.partial(PkgConfigDependency, 'shaderc', environment, kwargs))
+
+        if DependencyMethods.CMAKE in methods:
+            candidates.append(functools.partial(CMakeDependency, 'Shaderc', environment, kwargs))
+
+        if DependencyMethods.SYSTEM in methods:
+            candidates.append(functools.partial(ShadercDependency, environment, kwargs))
+
+        return candidates
+
+    @staticmethod
+    def get_methods():
+        return [DependencyMethods.PKGCONFIG, DependencyMethods.CMAKE, DependencyMethods.SYSTEM]
```

Comment:
I'm not sure that it's worth looking for pkg-config and cmake files honestly, it's just going to slow the process down. At the very least let's put SYSTEM first, and if they ever add pkg-config or cmake find modules we can reorder later. In fact, let's just reorder for now.